hera_pspec/uvwindow.py

```diff
@@ -0,0 +1,1320 @@
+                assert pols in [1, 2, 4, 3, -5, -6, -7, -8], "Wrong polarisation"
+                # convert pol number to str according to AIPS Memo 117.
+                pols = (uvputils.polnum2str(pols, x_orientation=x_orientation),
+                        uvputils.polnum2str(pols, x_orientation=x_orientation))      
+            else:
+                # if polarisation pair integer
+                pols = uvputils.polpair_int2tuple(pols, pol_strings=True)
+        # if tuple of polarisation strings
+        elif isinstance(pols[0], str):
+            assert pols[0] and pols[1] in ['pI', 'pQ', 'pV', 'pU', 'xx', 'yy', 'xy', 'yx'],\
+                "Wrong polarisation"
+        # if tuple of polarisation integers
+        elif isinstance(pols[0], int):
+            assert pols[0] and pols[1] in [1, 2, 4, 3, -5, -6, -7, -8],\
+                "Wrong polarisation"
+            pols = (uvutils.polnum2str(pols[0], x_orientation=x_orientation),
+                    uvutils.polnum2str(pols[0], x_orientation=x_orientation))
+        else:
+            raise TypeError("Must feed pols as (tuple of) str or int.")
+        self.pols = pols
+
+        # spectral window
+        if spw_range is not None:
+            # checks on spw range input
+            spw_range = np.array(spw_range, dtype=int)
+            assert spw_range.size == 2, "spw_range must be fed as a tuple of "\
+                                        "frequency indices."
+            assert spw_range[1]-spw_range[0] > 0 and min(spw_range) >= 0,\
+                "Require non-zero spectral range."
+
+        # create list of FTBeam objects for each polarisation channel
+        self.ftbeam_obj_pol = []
+        # if not fed, creates appropriate FTBeam objects
+        if ftbeam_obj is None:
+            # create ftbeam_obj for first pol in the pair
+            self.ftbeam_obj_pol.append(FTBeam(self.pols[0], ftfile=ftfile, spw_range=spw_range,
+                                              verbose=verbose))
+            # copy object if polarisations in pair are identical
+            if self.pols[0] == self.pols[1]:
+                self.ftbeam_obj_pol.append(copy.deepcopy(self.ftbeam_obj_pol[0]))
+            # else, create new ftbeam object
+            else:
+                self.ftbeam_obj_pol.append(FTBeam(self.pols[1], ftfile=ftfile, spw_range=spw_range,
+                                                  verbose=verbose))
+        # if fed, read inputs
+        else:
+            ftbeam_obj = ftbeam_obj if isinstance(ftbeam_obj, (list, tuple, np.ndarray)) else [ftbeam_obj, ftbeam_obj]
+            # if input is only one object
+            if (ftbeam_obj[0] == ftbeam_obj[1]):
+                # check if polarisations in pair are identical
+                assert ftbeam_obj[0].pol == self.pols[0] and ftbeam_obj[1].pol == self.pols[1],\
+                    'If feeding only one ftbeam_obj, polarisations in pair must be identical'
```

Comment:
using the above method of ONLY accepting the ftbeam_obj tuple would mean you don't need this check -- don't accept pols, just get them from the ftbeam_obj directly.